### ISSUE TITLE
psycopg2 error in home assistant v.2023.3.4

### ISSUE THREAD

**Issue by SergioRius** [freol35241/ltss#59]
```
Integration error: ltss - Requirements for ltss not found: ['psycopg2>=2.8,<3.0'].
```

Seems that is a not-required requirement. Could be?
The component seems to work normally without it.

**Comment by freol35241** [comment]
Hi, thanks for reporting, however I cant seem to reproduce this. Can you give some more details about your setup?

**Comment by freol35241** [comment]
Thanks for reporting back.

Well, sqlalchemy on its own does not know how to connect to a postgres instance without a separate package which can be anyone of the ones as specified [here](https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#dialect-postgresql).

And it looks like you already install [`psycopg2-binary`](https://github.com/tteck/Proxmox/blob/main/install/homeassistant-core-install.sh#L80) which would explain why it works without psycopg2 in the manifest file. Not sure if it is possible to install both psycopg2-binary and psycopg2 in the same environment?

**Comment by freol35241** [comment]
This issue may also have been made redundant with the merge of #42 ?

**Comment by SergioRius** [comment]
> This issue may also have been made redundant with the merge of #42 ?

Looks good to me.

**Comment by wizmo2** [comment]
Recommend reopening this in case others are experiencing the same issues.  PSYCOPG potentially has a fix in place (https://github.com/psycopg/psycopg2/commit/c96f991a8da73d9bebffe7fe9e69b191d36cceef) but not released yet (?).  In the meantime, as per @SergioRius, I edited the manifest.json to use pyscopg2 (removed '-binary') and it runs fine. 🥇 

I'm running on a 32bit RPI3 with
```
Home Assistant 2023.6.3
Supervisor 2023.06.4
Operating System 8.2 / 9.5/ 10.3
Frontend 20230608.0 - latest
```

And had this in the logs
```
2023-07-03 13:19:36.423 ERROR (SyncWorker_9) [homeassistant.util.package] Unable to install package psycopg2-binary>=2.8,<3.0: error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [37 lines of output]
      /usr/local/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
      !!
      
              ********************************************************************************
              The license_file parameter is deprecated, use license_files instead.
      
              By 2023-Oct-30, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
      
              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************
      
      !!
        parsed = self.parsers.get(option_name, lambda x: x)(value)
      running egg_info
      creating /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info
      writing /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/PKG-INFO
      writing dependency_links to /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/dependency_links.txt
      writing top-level names to /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/top_level.txt
      writing manifest file '/tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/SOURCES.txt'
      
      Error: pg_config executable not found.
      
      pg_config is required to build psycopg2 from source.  Please add the directory
      containing pg_config to the $PATH or specify the full executable path with the
      option:
      
          python setup.py build_ext --pg-config /path/to/pg_config build ...
      
      or with the pg_config option in 'setup.cfg'.
      
      If you prefer to avoid building psycopg2 from source, please install the PyPI
      'psycopg2-binary' package instead.
      
      For further information please check the 'doc/src/install.rst' file (also at
      <https://www.psycopg.org/docs/install.html>).
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
2023-07-03 13:19:36.429 ERROR (MainThread) [homeassistant.setup] Setup failed for custom integration ltss: Requirements for ltss not found: ['psycopg2-binary>=2.8,<3.0'].
```

**Comment by freol35241** [comment]
> Recommend reopening this in case others are experiencing the same issues. PSYCOPG potentially has a fix in place ([psycopg/psycopg2@c96f991](https://github.com/psycopg/psycopg2/commit/c96f991a8da73d9bebffe7fe9e69b191d36cceef)) but not released yet (?). In the meantime, as per @SergioRius, I edited the manifest.json to use pyscopg2 (removed '-binary') and it runs fine. 🥇
> 
> I'm running on a 32bit RPI3 with
> 
> ```
> Home Assistant 2023.6.3
> Supervisor 2023.06.4
> Operating System 8.2 / 9.5/ 10.3
> Frontend 20230608.0 - latest
> ```
> 
> And had this in the logs
> 
> ```
> 2023-07-03 13:19:36.423 ERROR (SyncWorker_9) [homeassistant.util.package] Unable to install package psycopg2-binary>=2.8,<3.0: error: subprocess-exited-with-error
>   
>   × python setup.py egg_info did not run successfully.
>   │ exit code: 1
>   ╰─> [37 lines of output]
>       /usr/local/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
>       !!
>       
>               ********************************************************************************
>               The license_file parameter is deprecated, use license_files instead.
>       
>               By 2023-Oct-30, you need to update your project and remove deprecated calls
>               or your builds will no longer be supported.
>       
>               See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
>               ********************************************************************************
>       
>       !!
>         parsed = self.parsers.get(option_name, lambda x: x)(value)
>       running egg_info
>       creating /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info
>       writing /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/PKG-INFO
>       writing dependency_links to /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/dependency_links.txt
>       writing top-level names to /tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/top_level.txt
>       writing manifest file '/tmp/pip-pip-egg-info-vx4_dr1h/psycopg2_binary.egg-info/SOURCES.txt'
>       
>       Error: pg_config executable not found.
>       
>       pg_config is required to build psycopg2 from source.  Please add the directory
>       containing pg_config to the $PATH or specify the full executable path with the
>       option:
>       
>           python setup.py build_ext --pg-config /path/to/pg_config build ...
>       
>       or with the pg_config option in 'setup.cfg'.
>       
>       If you prefer to avoid building psycopg2 from source, please install the PyPI
>       'psycopg2-binary' package instead.
>       
>       For further information please check the 'doc/src/install.rst' file (also at
>       <https://www.psycopg.org/docs/install.html>).
>       
>       [end of output]
>   
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> 
> × Encountered error while generating package metadata.
> ╰─> See above for output.
> 
> note: This is an issue with the package mentioned above, not pip.
> hint: See above for details.
> 2023-07-03 13:19:36.429 ERROR (MainThread) [homeassistant.setup] Setup failed for custom integration ltss: Requirements for ltss not found: ['psycopg2-binary>=2.8,<3.0'].
> ```

I have to say I am slightly confused here...

You get this error while trying to install psycopg2-binary? But the error message contains text suggesting that in order to avoid building psycopg2 from source, you should use psycopg2-binary? This suggests that there is no pre-built wheel-file available for your environment (rpi, python3.11) which also seems to be the case (https://www.piwheels.org/project/psycopg2-binary/). 

On the other hand, I assume you are running home assistant OS which I assumed made sure all necessary wheels are available for the supported platform, but my knowledge here is a bit rusty...

**Comment by wizmo2** [comment]
Me too! (on confused and rusty).   I assume its a Python 11 'thing'.  There seems to be a lot of them!   As you say the binary is not available, and the PR to fix the build issue has not been released.  My guess is that the HA team manually built the docker image from the psycopg2 current branch.  I'm not setup to debug on the RPi, and since this all goes away when the adapter gets a new release....
